sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/spice_indexer.py

```diff
@@ -0,0 +1,386 @@
+    try:
+        latest_lsk = furnish_best_spice_file(spice_file.parent.parent / "lsk")
+        latest_sclk = furnish_best_spice_file(spice_file.parent.parent / "sclk")
+    except FileNotFoundError as e:
+        if spice_metadata["type"] in ("leapseconds", "spacecraft_clock"):
+            # This block will likely only be reached if this is the very first
+            # leapsecond or spacecraft_clock kernel placed on the SDS. In this case,
+            # we'll insert default data and continue.
+            file_coverage_datetime = MAXIMUM_DATETIME_INTERVAL
+            file_coverage_j2000 = MAXIMUM_J2000_INTERVAL
+            file_coverage_sclk = MAXIMUM_SCLK_INTERVAL
+        else:
+            raise e
+
+    if latest_lsk and latest_sclk:  # clock and leapsecond kernels are loaded
+        if spice_metadata["start_date"] is None or spice_metadata["end_date"] is None:
+            # In this block, we have files that do NOT need to have
+            # any file_intervals calculated. We will use the maximum time range.
+            if spice_metadata["start_date"] is None:
+                spice_metadata["start_date"] = minimum_mission_time
+            if spice_metadata["end_date"] is None:
+                spice_metadata["end_date"] = maximum_mission_time
+            file_coverage_datetime = [
+                [spice_metadata["start_date"], spice_metadata["end_date"]]
+            ]
+            file_coverage_j2000 = [
+                [
+                    spiceypy.datetime2et(spice_metadata["start_date"]),
+                    spiceypy.datetime2et(spice_metadata["end_date"]),
+                ]
+            ]
+            file_coverage_sclk = [
+                [
+                    spiceypy.sce2s(SPACECRAFT_ID, file_coverage_j2000[0][0]),
+                    spiceypy.sce2s(SPACECRAFT_ID, file_coverage_j2000[0][1]),
+                ]
+            ]
+        else:
+            function_arguments = {
+                "idcode": SPACECRAFT_ID * 1000,
+                "cover": spiceypy.cell_double(COVERAGE_SPICE_ARRAY_LENGTH),
+            }
+            if "attitude" in spice_metadata["type"]:  # Extra arguments needed for ckcov
+                function_arguments["needav"] = COVERAGE_ANGULAR_VELOCITY_ONLY
+                function_arguments["level"] = COVERAGE_LEVEL
+                function_arguments["tol"] = COVERAGE_TOLERANCE
+                function_arguments["timsys"] = COVERAGE_TIME_SYSTEM
+            file_coverage_j2000, file_coverage_datetime, file_coverage_sclk = (
+                get_coverage_dictionary(spice_file, **function_arguments)
+            )
+
+    # Insert/Update the gathered data into the database
+    _upsert_into_spice_table(
+        spice_object,
+        file_coverage_j2000,
+        file_coverage_datetime,
+        file_coverage_sclk,
+        latest_lsk,
+        latest_sclk,
+    )
+
+
+def create_symlink(source_path: Path, destination_path: Path) -> None:
```

Comment:
I think this function is now unused now, can you remove it?

Reply:
Yup! I was wondering about that